plugin/src/main/scala/sbtcompendium/CompendiumPlugin.scala

```diff
@@ -22,29 +22,34 @@ import hammock.asynchttpclient.AsyncHttpClientInterpreter
 import higherkindness.compendium.models._
 import higherkindness.compendium.models.config.HttpConfig
 import sbt._
-import sbtcompendium.client.{CompendiumClient, CompendiumClientConfig}
 
 object CompendiumPlugin extends AutoPlugin {
 
   override def trigger: PluginTrigger = allRequirements
 
   object autoImport {
 
-    lazy val compendiumGenClients: TaskKey[Seq[File]] = taskKey[Seq[File]]("Generate all the clients for each protocol")
-
-    lazy val compendiumServerHost: SettingKey[String] = settingKey[String]("Url of the compendium server")
-    lazy val compendiumServerPort: SettingKey[Int]    = settingKey[Int]("Port of the compendium server")
+    import sbtcompendium.client.{CompendiumClient, CompendiumClientConfig}
+    lazy val compendiumGenClients: TaskKey[Seq[File]]   = taskKey[Seq[File]]("Generate all the clients for each protocol")
+    lazy val compendiumServerHost: SettingKey[String]   = settingKey[String]("Url of the compendium server")
+    lazy val compendiumServerPort: SettingKey[Int]      = settingKey[Int]("Port of the compendium server")
+    lazy val compendiumFormatSchema: SettingKey[String] = settingKey[String]("Schema type to download")
     lazy val compendiumProtocolIdentifiers: SettingKey[Seq[String]] =
       settingKey[Seq[String]]("Protocol identifiers to be retrieved from compendium server")
 
     def client(host: String, port: Int): CompendiumClient[IO] = {
-
-      implicit val interpreter = AsyncHttpClientInterpreter.instance[cats.effect.IO]
-      implicit val clientConfig: CompendiumClientConfig = CompendiumClientConfig(
-        HttpConfig(host, port)
-      )
+      implicit val interpreter                          = AsyncHttpClientInterpreter.instance[cats.effect.IO]
+      implicit val clientConfig: CompendiumClientConfig = CompendiumClientConfig(HttpConfig(host, port))
       CompendiumClient[cats.effect.IO]()
     }
+
+    def fromStringToIdlName(sch: String) = sch match {
```

Comment:
You wouldn't need this function if we directly receive the type through the `compendiumFormatSchema` sbt setting